src/metatrain/experimental/nanopet/model.py

```diff
@@ -706,6 +708,20 @@ def _add_output(self, target_name: str, target_info: TargetInfo) -> None:
     def upgrade_checkpoint(checkpoint: Dict) -> Dict:
         raise NotImplementedError("checkpoint upgrade is not implemented for NanoPET")
 
+    def _get_checkpoint(self) -> Dict:
```

Comment:
Not sure about the naming of this function with the underscore, since then we call it from outside the class (i.e. from the trainer) and therefore it's not acting as a private method